base/interactiveutil.jl

```diff
@@ -670,7 +670,7 @@ function summarysize(obj::Module, seen, excl)
     haskey(seen, obj) ? (return 0) : (seen[obj] = true)
     size::Int = Core.sizeof(obj)
     for binding in names(obj, true)
-        if isdefined(obj, binding)
+        if isdefined(obj, binding) && !isdeprecated(obj, binding)
```

Comment:
agreed. but we need some non-warning form of `getfield` so that we can reflect on them without causing warnings to be printed (and with `--depwarn=error` to fail). However, I'm not sure what that API should look